File: app/src/main/java/com/example/patiobalmax/adapter/PuestoAdapter.java
package com.example.patiobalmax.adapter;

import android.content.Context;
import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.patiobalmax.R;
import com.example.patiobalmax.database.PuestoEntity;
import com.example.patiobalmax.model.EstadoPuesto;
import com.example.patiobalmax.ui.EditarPatioEstacionamiento;

import java.util.List;

public class PuestoAdapter extends RecyclerView.Adapter<PuestoAdapter.PuestoViewHolder> {

    private List<PuestoEntity> puestos;
    private OnItemClickListener listener;
    private Context context;

    public interface OnItemClickListener {
        void onItemClick(PuestoEntity puesto);
    }

    public PuestoAdapter(List<PuestoEntity> puestos, OnItemClickListener listener) {
        this.puestos = puestos;
        this.listener = listener;
    }

    @NonNull
    @Override
    public PuestoViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        context = parent.getContext();
        View view = LayoutInflater.from(context).inflate(R.layout.item_puesto, parent, false);
        return new PuestoViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull PuestoViewHolder holder, int position) {
        PuestoEntity puesto = puestos.get(position);
        holder.tvNumero.setText("Puesto " + (position + 1));
        holder.tvEstado.setText(obtenerEstado(puesto));

        // Asignar color según estado
        EstadoPuesto estado = EstadoPuesto.valueOf(puesto.estado);
        switch (estado) {
            case ARRENDATARIO:
                holder.itemView.setBackgroundColor(Color.parseColor("#4CAF50")); // Verde
                break;
            case PARTICULAR:
                holder.itemView.setBackgroundColor(Color.parseColor("#2196F3")); // Azul
                break;
            default:
                holder.itemView.setBackgroundColor(Color.parseColor("#FFFFFF")); // Blanco
                break;
        }

        // Asignar icono según tipo de vehículo
        String tipoVehiculo = puesto.tipoVehiculoLugar1;
        if (tipoVehiculo.contains("Camión") || tipoVehiculo.contains("Maquinaria")) {
            holder.ivIcono.setImageResource(R.drawable.icono_camion);
        } else {
            holder.ivIcono.setImageResource(R.drawable.icono_auto);
        }

        // Mostrar icono de ticket si tiene permiso
        holder.ivTicket.setVisibility(View.GONE);
        holder.ivEdit.setVisibility(View.GONE);
        holder.itemView.setOnClickListener(v -> listener.onItemClick(puesto));
    }

    private String obtenerEstado(PuestoEntity puesto) {
        return puesto.estado;
    }

    @Override
    public int getItemCount() {
        return puestos.size();
    }

    static class PuestoViewHolder extends RecyclerView.ViewHolder {
        TextView tvNumero, tvEstado;
        ImageView ivIcono, ivTicket, ivEdit;

        public PuestoViewHolder(@NonNull View itemView) {
            super(itemView);
            tvNumero = itemView.findViewById(R.id.tvNumeroPuesto);
            tvEstado = itemView.findViewById(R.id.tvEstado);
            ivIcono = itemView.findViewById(R.id.ivIconoVehiculo);
            ivTicket = itemView.findViewById(R.id.ivTicket);
            ivEdit = itemView.findViewById(R.id.ivEdit);
        }
    }
}

File: app/src/main/kotlin/com/example/patiobalmax/adapter/PuestoAdapter.kt
package com.example.patiobalmax.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.example.patiobalmax.databinding.ItemPuestoBinding
import com.example.patiobalmax.database.entity.Puesto

class PuestoAdapter(val onClick: (Puesto) -> Unit) :
    ListAdapter<Puesto, PuestoAdapter.PuestoViewHolder>(PuestoDiffCallback()) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PuestoViewHolder {
        val binding = ItemPuestoBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return PuestoViewHolder(binding)
    }

    override fun onBindViewHolder(holder: PuestoViewHolder, position: Int) {
        holder.bind(getItem(position))
    }

    inner class PuestoViewHolder(private val binding: ItemPuestoBinding) :
        RecyclerView.ViewHolder(binding.root) {

        fun bind(puesto: Puesto) {
            binding.tvNumeroPuesto.text = "Puesto ${puesto.numeroPuesto}"
            binding.etPlacaLugar1.setText(puesto.placa1)
            binding.etPlacaLugar2.setText(puesto.placa2)

            when {
                puesto.estado == "Arrendatario" -> binding.root.setBackgroundColor(itemView.resources.getColor(android.R.color.holo_green_light))
                puesto.estado == "Particular" -> binding.root.setBackgroundColor(itemView.resources.getColor(android.R.color.holo_blue_light))
                else -> binding.root.setBackgroundColor(itemView.resources.getColor(android.R.color.white))
            }

            binding.iconoEdit.setOnClickListener {
                onClick(puesto)
            }
        }
    }
}

class PuestoDiffCallback : DiffUtil.ItemCallback<Puesto>() {
    override fun areItemsTheSame(oldItem: Puesto, newItem: Puesto): Boolean =
        oldItem.id == newItem.id

    override fun areContentsTheSame(oldItem: Puesto, newItem: Puesto): Boolean =
        oldItem == newItem
}
